manifests/compute.pp

```diff
@@ -1,9 +1,37 @@
 # @summary Setup slurm config for slurm compute node
 #
+# @param dependencies
+#   Optionally list resources (e.g. mounts) that should be present before
+#   setting up Slurm on a compute node. Should be in the form that would
+#   be specified as a requirement ("before") various Slurm compute
+#   resources, e.g.:
+#     "Gpfs::Bindmount['/scratch']"
+#     "Gpfs::Bindmount['/sw']"
+#     "Gpfs::Bindmount['/u']"
+#     "Gpfs::Nativemount['/cluster']"
+#     "Lvm::Logical_volume[local]"
+#     "Profile_lustre::Nativemount_resource['/projects']"
+#     "Profile_lustre::Nativemount_resource['/taiga']"
+#
+# @param require_dependencies
+#   Optionally list resources (e.g., services) that require any of the
+#   items from the $dependencies parameter in order to function.
```

Comment:
The name of `require_dependencies` is a little confusing. If I was just glancing though hiera the first thing that would come to mind when seeing that param would be "list of dependencies required to make the module work". Then if I saw the regular dependencies in `profile_slurm::compute::dependencies:` I'd get pretty confused.

Granted you do a good job explaining the difference between the dependencies in the code comments, but if the param names were changed so you could figure out the differences just based of their names that would be a big bonus.

Maybe we keep `dependencies` and then rename the other one to `require_after_dependencies`. Which would make it clear that we have our normal dependencies list, and then a second layer of dependencies that should happen after the normal ones.

Reply:
Per discussion in Slack I changed `require_dependencies` to `configure_after_dependencies`. I retested and it seems all good.